src/lib/contracts.ts

```diff
@@ -51,13 +94,30 @@ function isValidRequirementType(
 	return requirement in requirementVersions;
 }
 
+function partition<T>(arr: T[], predicate: (t: T) => boolean) {
+	return arr.reduce(
+		({ pass, fail }, elem) => {
+			if (predicate(elem)) {
+				pass.push(elem);
+			} else {
+				fail.push(elem);
+			}
+			return { pass, fail };
+		},
+		{ pass: [] as T[], fail: [] as T[] },
+	);
+}
+
 export function containerContractsFulfilled(
-	serviceContracts: ServiceContracts,
-): ApplicationContractResult {
-	const containers = _(serviceContracts).map('contract').compact().value();
+	servicesWithContract: ServiceWithContract[],
+): AppContractResult {
+	const containers = servicesWithContract
+		.map(({ contract }) => contract)
+		.filter((c) => !!c) as ContractObject[];
```

Comment:
As a general rule it's better to do `satisfies X` rather than `as X`, or `satisfies X as X` so that it first checks that the inferred typing is compatible with the expected typing before doing the cast and increase the type safety